arm/ripper/handbrake.py

```diff
@@ -26,6 +29,22 @@ def handbrake_mainfeature(srcpath, basepath, logfile, job):
     """
     logging.info("Starting DVD Movie Mainfeature processing")
     logging.debug("Handbrake starting: " + str(job))
+    
+    # Wait until there is a spot to transcode
+    job.status = "waiting_transcode"
+    logging.info("Setting job status to 'waiting_transcode' and checking to see if there is a slot")
+    transcode_number = 9999999
+    while transcode_number >= job.config.MAX_CONCURRENT_TRANSCODES:
+        #Count how many times handbrake is currently running
+        #procs = subprocess.check_output(['ps', '-a', '-c', '-ocomm=']).splitlines()
+        #TRANSCODE-NUMBER = procs.count('HandBrakeCLI')
+        transcode_number = sum(1 for proc in psutil.process_iter() if proc.name() == 'HandBrakeCLI')
+        logging.info("Number of HandBrakeCLI Processes running is:" + str(transcode_number) + " going to waiting 12 seconds.")
+        time.sleep(12)
+    
+    logging.debug("Setting job status to 'transcoding'")
+    job.status = "transcoding"
+    
```

Comment:
Could this be extracted into a method/function? It just seems like a code smell to me that there's a duplicate here.